src/linux/cgroups2.cpp

```diff
@@ -465,6 +465,53 @@ const std::string UCLAMP_MIN = "cpu.uclamp.min";
 const std::string WEIGHT = "cpu.weight";
 const std::string WEIGHT_NICE = "cpu.weight.nice";
 
+namespace stat {
+
+Try<Stats> parse(const string& content)
+{
+  const vector<string>& lines = strings::split(content, "\n");
+  cpu::Stats stats;
+
+  foreach (const string& line, lines) {
+    const string trimmed = strings::trim(line);
+    if (trimmed.empty()) {
+      continue;
+    }
+
+    vector<string> tokens = strings::split(trimmed, " ");
+    if (tokens.size() != 2) {
+      return Error(
+          "Invalid line format in 'cpu.stat' expected "
+          "<key> <value> received: '" + trimmed + "'");
+    }
+
+    string field = tokens[0];
+    string value = tokens[1];
+
+    Try<Duration> duration = Duration::parse(value + "us");
```

Comment:
we can just construct Duration using `Duration d = Microseconds(static_cast<int64_t>(*number))`